pkg/controller/controllercmd/builder.go

```diff
@@ -31,9 +32,20 @@ type StartFunc func(*ControllerContext) error
 
 type ControllerContext struct {
 	ComponentConfig *unstructured.Unstructured
-	KubeConfig      *rest.Config
-	EventRecorder   events.Recorder
-	Context         context.Context
+
+	// KubeConfig provides the REST config with no content type (it will default to JSON).
+	// Use this config for CR resources.
+	KubeConfig *rest.Config
+
+	// ProtoKubeConfig provides the REST config with "application/vnd.kubernetes.protobuf,application/json" content type.
+	// Note that this config might not be safe for CR resources, instead it should be used for other resources.
+	ProtoKubeConfig *rest.Config
```

Comment:
@deads2k the `GetKubeConfigOrInClusterConfig` seems to already have some logic to prefer protobuf via overrides if the `Accept` is not set.

However, in prometheus we see the operator making a lot of requests for secrets/configs via JSON, which might suggest the default is set to JSON in client-go?

In any case, I like to separate proto vs. json kube configs at least until @smarterclayton fix problems with CRs.

Reply:
afaik default is still json in client-go. Each binary has to override that manually.

Reply:
@sttts in that case this good change then. the callers can pick which client they prefer depending on what resources are they going to deal with.